lib/emque/producing/message/message.rb

```diff
@@ -36,6 +36,16 @@ def read_raise_on_failure
           end
         end
 
+        def ignored_exceptions(*ignored_exceptions)
+          @ignored_exceptions = ignored_exceptions
+        end
+
+        def read_ignored_exceptions
+          (Array(@ignored_exceptions) +
+            [InvalidMessageError, MessagesNotSentError] +
```

Comment:
If we want to keep the original behavior we would include `MessagesNotSentError` because that matches the original behavior where we would always throw the `InvalidMessageError` but only throw `MessageNotSentError` if `raise_on_error?` was true.